File: srv/models/journal.py
from typing import List
from sqlmodel import JSON, Column, Field, SQLModel, Relationship
from datetime import datetime
from .user import User

def getDateTime() -> datetime:
    return datetime.now()

class Journal(SQLModel, table=True):
    id: int | None = Field(default=None, primary_key=True)
    user_id: int = Field(foreign_key="user.id", index=True)
    date: datetime = Field(default_factory=getDateTime, index=True)
    content:  List[str]  = Field(
        sa_column=Column(JSON),         # ← tell SQLAlchemy to use a JSON column
        default_factory=list,           # ← default to an empty list
    )

File: frontend/app/model/journal.ts
export class JournalData {
    date: string;
    id: number;
    user_id: number;
    content: string[];
    constructor(id: number, user_id: number, content: [string], date: string) {
        this.id = id;
        this.user_id = user_id;
        this.content = content;
        this.date = date;
    }
}
